src/main/java/io/cdap/plugin/snowflake/source/batch/SnowflakeBatchSource.java

```diff
@@ -88,6 +90,8 @@ public void prepareRun(BatchSourceContext context) {
     }
 
     context.setInput(Input.of(config.getReferenceName(), new SnowflakeInputFormatProvider(config, escapeChar)));
+    // set error details provider
+    context.setErrorDetailsProvider(new ErrorDetailsProviderSpec(SnowflakeErrorDetailsProvider.class.getName()));
```

Comment:
this should be done before `context.setInput(Input.of(config.getReferenceName(), new SnowflakeInputFormatProvider(config, escapeChar)))`